test/unit/deadline_client/ui/test_host_requirements_dataclass_serialize.py

```diff
@@ -0,0 +1,178 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+"""
+tests the deadline.client.ui.dataclasses functions relating to serializing.
+"""
+
+from deadline.client.ui.dataclasses import (
+    HostRequirements,
+    OsRequirements,
+    HardwareRequirements,
+    CustomRequirements,
+    CustomAmountRequirement,
+    CustomAttributeRequirement,
+)
+
+
+HOST_REQUIREMENTS_1 = {
+    "attributes": [
+        {"name": "attr.worker.os.family", "anyOf": ["windows"]},
+        {"name": "attr.worker.cpu.arch", "anyOf": ["x86_64"]},
+        {"name": "attr.worker.pipelineFeatures", "anyOf": ["feature1", "feature2"]},
+    ],
+    "amounts": [
+        {"name": "amount.worker.vcpu", "min": 8, "max": 64},
+        {"name": "amount.worker.memory", "min": 16384, "max": 131072},
+        {"name": "amount.worker.Bugs", "min": 1, "max": 10},
+    ],
+}
+
+
+def _compare_requirements(first, second):
```

Comment:
A cool, simple way to do this could be:
```
sorted_first_attributes = sorted(first.get("attributes". []), key=lambda x: x["name"])
sorted_second_attributes = sorted(second.get("attributes". []), key=lambda x: x["name"])
assert sorted_first_attributes == sorted_second_attributes

sorted_first_amounts = sorted(first.get("amounts". []), key=lambda x: x["name"])
sorted_second_amounts = sorted(second.get("amounts". []), key=lambda x: x["name"])
assert sorted_first_amounts == sorted_second_amounts
```